src/regmod/parameter.py

```diff
@@ -134,6 +93,8 @@ def get_mat(self, data: Data) -> np.ndarray:
         np.ndarray
             The design matrix.
         """
+        if len(self.variables) == 0:
+            return np.empty(shape=(data.df.shape[0], 0))
```

Comment:
Why do you need to create an empty matrix; why not just return nothing?

Reply:
This is for some calculation that combine the matrices or doing the dot product. I want it to naturally propagate through rather than at each step we need to check it is `None` or not. If there is better ideas please let me know :)